### PR TITLE
Added chip component

### PR DESCRIPTION
**Pull request by rawi-coding**
Added chip component: https://digital.sbb.ch/de/design-system/mobile/components/chip/

### CODE REVIEW

File: lib/src/chip/sbb_chip.dart [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  /// if [selected] is true, the unselect button is shown.
+  final bool selected;
+
+  @override
+  Widget build(BuildContext context) {
+    final style = SBBControlStyles.of(context).chip!;
+    return GestureDetector(
+      onTap: _disabled ? null : () => _changeSelection(true),
+      child: Container(
+        decoration: BoxDecoration(
```
Comment (VanHoangTran) [inline lib/src/chip/sbb_chip.dart:42]: StadiumBorder scales with bigger font sizes:
```dart
decoration: ShapeDecoration(
  shape: StadiumBorder(
    color: _disabled ? style.disabledBorderColor! : style.borderColor!,
  ),
),
```

File: lib/src/chip/sbb_chip.dart [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  }
+
+  Container _roundedContainer({
+    required Widget child,
+    required Color color,
+    double? width,
+    EdgeInsetsGeometry? padding,
+  }) {
+    return Container(
+      margin: EdgeInsets.all(4.0).copyWith(left: 0),
```
Comment (VanHoangTran) [inline lib/src/chip/sbb_chip.dart:105]: `left: 0` results in an asymmetric InkWell ripple effect.
Suggestion: Keep left margin of 4 here and subtract 4 from label right padding.

File: lib/src/theme/styles/src/sbb_chip_style.dart [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      unselectButtonColor: isDarkTheme ? SBBColors.iron : SBBColors.milk,
+    );
+  }
+
+  final Color? borderColor;
+  final Color? disabledBorderColor;
+  final Color? badgeColor;
+  final Color? disabledBadgeColor;
+  final Color? badgeTextColor;
+  final Color? unselectButtonColor;
```
Comment (VanHoangTran) [inline lib/src/theme/styles/src/sbb_chip_style.dart:32]: Missing text color for disabled state. I would also add text color for enabled state to be consistent.

File: lib/src/chip/sbb_chip.dart [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+          ],
+        ),
+      ),
+    );
+  }
+
+  Padding _label() {
+    return Padding(
+      padding: const EdgeInsets.symmetric(
+        horizontal: 12.0,
```
Comment (VanHoangTran) [inline lib/src/chip/sbb_chip.dart:65]: See my comment on line 105

**Review by VanHoangTran — CHANGES_REQUESTED** [review]
Looks pretty good so far. Needs some changes though ;-)

**Comment by rawi-coding** [comment]
@VanHoangTran I've updated the PR. Thanks for the review :-)

File: lib/src/theme/styles/src/sbb_chip_style.dart (hunk 3 above)
Comment (VanHoangTran) [reply lib/src/theme/styles/src/sbb_chip_style.dart:41]: Still missing label text color for both states ;-)

**Review by VanHoangTran — APPROVED** [review]

**Pull request merged**